### PR TITLE
[Feature] Improve interacive preview

### PR DESCRIPTION
**Pull request by moshuying**
Re: #
https://github.com/cocos/3d-tasks/issues/15470
https://github.com/cocos/3d-tasks/issues/14387
### Changelog

* Improve interacive preview

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by dogeFu — APPROVED** [review]

File: editor/inspector/assets/fbx/preview.js [hunk 1]
```diff
@@ -690,7 +707,7 @@ exports.methods = {
     },
 };
 
-exports.ready = function() {
+exports.ready = function () {
```
Comment (arsen2010) [inline editor/inspector/assets/fbx/preview.js]: enable eslint to formate

Comment (moshuying) [reply editor/inspector/assets/fbx/preview.js]: ok,formate fixed

**Review by arsen2010 — APPROVED** [review]

**Review by arsen2010 — APPROVED** [review]

File: editor/inspector/assets/fbx/preview.js [hunk 2]
```diff
@@ -339,7 +339,15 @@ const Elements = {
         ready() {
             const panel = this;
             panel.$.canvas.addEventListener('mousedown', async (event) => {
-                await callModelPreviewFunction('onMouseDown', { x: event.x, y: event.y });
+                await callModelPreviewFunction('onMouseDown', { x: event.x, y: event.y, button: event.button });
+
+                async function onkeydown(event) {
```
Comment (VisualSJ) [inline editor/inspector/assets/fbx/preview.js]: keydown

File: editor/inspector/assets/fbx/preview.js [hunk 3]
```diff
@@ -358,16 +366,25 @@ const Elements = {
...
                 document.addEventListener('mousemove', mousemove);
                 document.addEventListener('mouseup', mouseup);
+                document.addEventListener('keydown', onkeydown);
 
                 panel.isPreviewDataDirty = true;
             });
 
+            panel.$.canvas.addEventListener('wheel', async (event) => {
+                await callModelPreviewFunction('onMouseWheel', {
+                    wheelDeltaY: event.wheelDeltaY,
```
Comment (VisualSJ) [inline editor/inspector/assets/fbx/preview.js:373]: event.wheelDeltaX

File: editor/inspector/assets/material-header.js [hunk 4]
```diff
@@ -92,9 +92,10 @@ exports.methods = {
     },
 };
 
-exports.ready = async function() {
+exports.ready = async function () {
```
Comment (VisualSJ) [inline editor/inspector/assets/material-header.js]: space？

File: editor/inspector/assets/material-header.js [hunk 5]
```diff
@@ -108,7 +109,15 @@ exports.ready = async function() {
     });
 
     panel.$.canvas.addEventListener('mousedown', async (event) => {
-        await callMaterialPreviewFunction('onMouseDown', { x: event.x, y: event.y });
+        await callMaterialPreviewFunction('onMouseDown', { x: event.x, y: event.y, button: event.button });
+
+        async function onkeydown(event) {
```
Comment (VisualSJ) [inline editor/inspector/assets/material-header.js]: keydown

File: editor/inspector/assets/material-header.js [hunk 6]
```diff
@@ -127,15 +136,24 @@ exports.ready = async function() {
...
         document.addEventListener('mousemove', mousemove);
         document.addEventListener('mouseup', mouseup);
+        document.addEventListener('keydown', onkeydown);
 
         panel.isPreviewDataDirty = true;
     });
 
+    panel.$.canvas.addEventListener('wheel', async (event) => {
+        await callMaterialPreviewFunction('onMouseWheel', {
+            wheelDeltaY: event.wheelDeltaY,
```
Comment (VisualSJ) [inline editor/inspector/assets/material-header.js:142]: x

File: editor/inspector/assets/skeleton-preview.js [hunk 7]
```diff
@@ -1,5 +1,7 @@
 'use strict';
 
+const { doc } = require("prettier");
```
Comment (VisualSJ) [inline editor/inspector/assets/skeleton-preview.js]: " --> '

Comment (VisualSJ) [reply editor/inspector/assets/skeleton-preview.js]: require?

File: editor/inspector/assets/fbx/preview.js (hunk 2 above)
Comment (moshuying) [reply editor/inspector/assets/fbx/preview.js]: ok

File: editor/inspector/assets/fbx/preview.js (hunk 3 above)
Comment (moshuying) [reply editor/inspector/assets/fbx/preview.js:373]: ok

File: editor/inspector/assets/skeleton-preview.js (hunk 7 above)
Comment (moshuying) [reply editor/inspector/assets/skeleton-preview.js]: ok , i was delete it

File: editor/inspector/assets/material-header.js (hunk 6 above)
Comment (moshuying) [reply editor/inspector/assets/material-header.js:142]: OK

File: editor/inspector/assets/material-header.js (hunk 5 above)
Comment (moshuying) [reply editor/inspector/assets/material-header.js]: OK

File: editor/inspector/assets/material-header.js (hunk 4 above)
Comment (moshuying) [reply editor/inspector/assets/material-header.js]: fixed

File: editor/inspector/assets/fbx/preview.js [hunk 8]
```diff
@@ -339,7 +339,15 @@ const Elements = {
         ready() {
             const panel = this;
             panel.$.canvas.addEventListener('mousedown', async (event) => {
-                await callModelPreviewFunction('onMouseDown', { x: event.x, y: event.y });
+                await callModelPreviewFunction('onMouseDown', { x: event.x, y: event.y, button: event.button });
+
+                async function keydown(event) {
+                    await callModelPreviewFunction('onKeyDown', {
```
Comment (VisualSJ) [inline editor/inspector/assets/fbx/preview.js]: 1. Shouldn't a generic keydown event be triggered after mousedown?
2. If it's an event on the document, should we still use 'onKeyDown'? What should we name it if we need to listen to events on the canvas in the future?

**Review by VisualSJ — APPROVED** [review]

**Pull request merged**